### PR TITLE
Add note for alias case sensitivity

### PR DESCRIPTION
**Pull request by B-Step62**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/B-Step62/mlflow/pull/12344?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/12344/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 12344
```

</p>
</details>

### Related Issues/PRs

https://github.com/mlflow/mlflow/issues/12331

### What changes are proposed in this pull request?

Model alias is not case sensitive by design. Adding that information to the documentation.

### How is this PR tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [x] Manual tests

### Does this PR require documentation update?

- [x] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [x] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

`Yes` should be selected for bug fixes, documentation updates, and other small changes. `No` should be selected for new features and larger changes. If you're unsure about the release classification of this PR, leave this unchecked to let the maintainers decide.

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Bug fixes, doc updates and new features usually go into minor releases.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Bug fixes and doc updates usually go into patch releases.

</details>

- [ ] Yes (this PR will be cherry-picked and included in the next patch release)
- [x] No (this PR will be included in the next minor release)

### CODE REVIEW

**Comment by harupy** [comment]
Not sure if this is intended but UI lowercases aliases:

https://github.com/mlflow/mlflow/assets/17039389/f0840b1f-e4db-4691-a560-758c984bb20c

I'm wondering why we need to lowercase aliases if they are case-insensitive.

**Comment by B-Step62** [comment]
> I'm wondering why we need to lowercase aliases if they are case-insensitive.

Hmm MLflow actually lowercases the alias even if we set it via API `set_registered_model_alias`. So doing so on UI makes sense to avoid confusion.

Maybe better wording is `lowercased` rather than `not case-sensitive`?

**Comment by harupy** [comment]
The following code doesn't lowercase the HELLO alias:

```python
import mlflow

import mlflow.sklearn
from mlflow import MlflowClient

client = MlflowClient()

model_name = "sk-learn-random-forest-reg-model"
with mlflow.start_run():
    mlflow.sklearn.log_model({}, "model", registered_model_name=model_name)

# Set model version alias
model_version_alias = "HELLO"
client.set_registered_model_alias(model_name, model_version_alias, "1")

```

**Comment by B-Step62** [comment]
@harupy wow, yea I tested in Databricks actually, and it does lowercase

```
import mlflow
from mlflow import MlflowClient

mlflow.set_registry_uri("databricks-uc")

client = MlflowClient()
client.set_registered_model_alias("yuki_watanabe_test.mlflow_tracing.lc_serving_39_bug_bash", "TEST", 2)
```

And this works
```
client.get_model_version_by_alias("yuki_watanabe_test.mlflow_tracing.lc_serving_39_bug_bash", "test")
```

Could you check if `get_model_version_by_alias(model_name, "test")` works on your example? If it does, that means alias is also case-insensitive in OSS.

**Comment by harupy** [comment]
`get_model_version_by_alias(model_name, "hello")` works because MacOS is not a case sensitive file system lol.

File: docs/source/model-registry.rst [hunk 1]
```diff
@@ -34,6 +34,10 @@ Model Alias
 
     Aliases are especially useful for deploying models. For example, you could assign a ``champion`` alias to the model version intended for production traffic and target this alias in production workloads. You can then update the model serving production traffic by reassigning the ``champion`` alias to a different model version.
 
+    .. note::
+
+        Model Alias is not case-sensitive. For example, ``champion`` and ``Champion`` are both converted to lowercase and considered the same alias.
```
Comment (harupy) [inline docs/source/model-registry.rst:39]: so this is databricks-only behavior?

**Comment by B-Step62** [comment]
> get_model_version_by_alias(model_name, "hello") works because MacOS is not a case sensitive file system lol (I'm using file store). In sqlalchemy store, it doesn't work.

Ooh interesting, so this is Databricks only behavior and then we should fix it in OSS frontend.

File: docs/source/model-registry.rst (hunk 1 above)
Comment (BalkanFlink) [reply docs/source/model-registry.rst:39]: [It seems so](https://github.com/mlflow/mlflow/issues/12331#issuecomment-2164959769)

**Pull request closed without merging**